### PR TITLE
rl_chain.Embed used in base chain and in pick best chain

### PR DESCRIPTION
**Pull request by olgavrou**
also added some unit tests and a github action

working towards issue #3

### CODE REVIEW

File: rl_chain/rl_chain_base.py [hunk 1]
```diff
@@ -199,55 +201,53 @@ def _learn(self, vw_ex):
         self.workspace.learn_one(multi_ex)
 
 
+def embed_string_type(item: Union[str, _Embed], model: Any, default_namespace: Optional[str] = None) -> Dict[str, str]:
```
Comment (ataymano) [inline rl_chain/rl_chain_base.py:204]: looks like default_namespace is not needed anymore?

Comment (ataymano) [reply rl_chain/rl_chain_base.py:204]: sorry, it is still used, but it is not "default" anymore

Comment (olgavrou) [reply rl_chain/rl_chain_base.py:204]: maybe should be renamed to `namespace`

**Review by ataymano — APPROVED** [review]

File: rl_chain/rl_chain_base.py (hunk 1 above)
Comment (ataymano) [reply rl_chain/rl_chain_base.py:204]: yeah, there is no alternative to it anyway

Comment (olgavrou) [reply rl_chain/rl_chain_base.py]: kay done

File: rl_chain/rl_chain_base.py [hunk 2]
```diff
@@ -199,55 +201,53 @@ def _learn(self, vw_ex):
...
+def embed_dict_type(item: Dict, model: Any) -> Dict[str, str]:
+    """Helper function to embed a dictionary item."""
+    inner_dict = {}
+    for ns, embed_str in item.items():
+        inner_dict.update(embed_string_type(embed_str, model, ns))
+    return inner_dict
+
 def embed(
-    to_embed: Union[str, Dict, List[str], List[Dict]],
+    to_embed: Union[Union(str, _Embed(str)), Dict, List[Union(str, _Embed(str))], List[Dict]],
```
Comment (ataymano) [inline rl_chain/rl_chain_base.py:226]: [nit] looks splitting into _embed(Union[str, _Embed(str), Dict]) and embed(...) can simplify the code (we just call _embed in list case without caring about type) - but maybe not very much

**Pull request merged**